core/src/main/java/org/elasticsearch/indices/recovery/PeerRecoveryTargetService.java

```diff
@@ -169,50 +175,15 @@ private void doRecovery(final long recoveryId) {
                 logger.trace("not running recovery with id [{}] - can't find it (probably finished)", recoveryId);
                 return;
             }
-            RecoveryTarget recoveryTarget = recoveryRef.target();
+            final RecoveryTarget recoveryTarget = recoveryRef.target();
             assert recoveryTarget.sourceNode() != null : "can't do a recovery without a source node";
 
-            logger.trace("collecting local files for {}", recoveryTarget.sourceNode());
-            Store.MetadataSnapshot metadataSnapshot;
-            try {
-                if (recoveryTarget.indexShard().indexSettings().isOnSharedFilesystem()) {
-                    // we are not going to copy any files, so don't bother listing files, potentially running
-                    // into concurrency issues with the primary changing files underneath us.
-                    metadataSnapshot = Store.MetadataSnapshot.EMPTY;
-                } else {
-                    metadataSnapshot = recoveryTarget.indexShard().snapshotStoreMetadata();
-                }
-                logger.trace("{} local file count: [{}]", recoveryTarget, metadataSnapshot.size());
-            } catch (org.apache.lucene.index.IndexNotFoundException e) {
-                // happens on an empty folder. no need to log
-                logger.trace("{} shard folder empty, recover all files", recoveryTarget);
-                metadataSnapshot = Store.MetadataSnapshot.EMPTY;
-            } catch (IOException e) {
-                logger.warn("error while listing local files, recover as if there are none", e);
-                metadataSnapshot = Store.MetadataSnapshot.EMPTY;
-            } catch (Exception e) {
-                // this will be logged as warning later on...
-                logger.trace("unexpected error while listing local files, failing recovery", e);
-                onGoingRecoveries.failRecovery(recoveryTarget.recoveryId(),
-                    new RecoveryFailedException(recoveryTarget.state(), "failed to list local files", e), true);
-                return;
-            }
+            final Optional<StartRecoveryRequest> maybeRequest = getStartRecoveryRequest(recoveryTarget);
+            if (!maybeRequest.isPresent()) return;
```

Comment:
Okay, that's exactly what I did *except* I was trying to preserve the messages [here](https://github.com/elastic/elasticsearch/blob/ee5f8c452264d47181600d00427b32d08b0d60e2/core/src/main/java/org/elasticsearch/indices/recovery/PeerRecoveryTargetService.java#L197) and [here](https://github.com/elastic/elasticsearch/blob/ee5f8c452264d47181600d00427b32d08b0d60e2/core/src/main/java/org/elasticsearch/indices/recovery/PeerRecoveryTargetService.java#L213), and that it uglier than I preferred. If you're okay dropping those and just getting a generic message, and you appear to be, then I'll just do that already. :smile: